Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialUtils.cpp

```diff
@@ -226,6 +227,40 @@ namespace AZ
                 // the string (for the extension) we can assume it's an Image file path.
                 return value.Is<AZStd::string>() && AzFramework::StringFunc::Contains(value.GetValue<AZStd::string>(), ".");
             }
+
+            bool IsValidName(AZStd::string_view name)
+            {
+                // Checks for a c++ style identifier
+                return AZStd::regex_match(name.begin(), name.end(), AZStd::regex("^[a-zA-Z_][a-zA-Z0-9_]*$"));
```

Comment:
In some regex engines, it's important to cache the state machine constructed from parsing the regex string. Is that the case here?

Reply:
I searched the code for other uses of regex_match and they're all doing the same thing.

Reply:
Looking at the code of `AZStd::regex`, it does appear to compile the regex every single time, so if we're doing this everywhere that's unfortunate. In this case the string is short and the parsing time probably isn't too bad, but maybe something to be aware of.